File: MergeSortedArray.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    var j=0,k=0;
    var nums11=nums1.slice(0, m); 
    for(var i=0;i<m+n;i++)
    {
        if(j>=m)
        {
            nums1[i]=nums2[k];
            k++;
        
        }
        else if(k>=n)
        {
            nums1[i]=nums11[j];
            j++;
        
        }
        else if(nums11[j]>=nums2[k])
        {
            nums1[i]=nums2[k];
            k++;
        }
        else
        {
            nums1[i]=nums11[j];
            j++;
        }
    }

};

File: MergeSortedArray.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) 
    {
        
        int i,j,k;
        vector<int>temp1=nums1;
    
        i=k=j=0;
        while(k<n+m)
        {
            if(j>=n)
            {
                nums1[k++]=temp1[i++];
                continue;
            }
            if(i>=m)
            {
                nums1[k++]=nums2[j++];
                continue;
            }
            if(temp1[i]<=nums2[j])
            {
                nums1[k++]=temp1[i++];
            }
            else
            {
                nums1[k++]=nums2[j++];
            }
            
        }
        
    }
    
        

};

int main()
{
    
    return 0;
}
